lib/core/covfie/core/algebra/vector.hpp

```diff
@@ -42,8 +42,9 @@ struct vector : public matrix<N, 1, T, I> {
     template <
         typename... Args,
         std::enable_if_t<
-            (std::is_scalar_v<Args> && ...) &&
-                (std::is_convertible_v<Args, T> && ...) && sizeof...(Args) == N,
+            std::conjunction_v<std::is_scalar<Args>...> &&
+                std::conjunction_v<std::is_convertible<Args, T>...> &&
+                sizeof...(Args) == N,
```

Comment:
Similar situation here. In C\+\+17 mode this fails with:

```
C:\Users\krasz\ATLAS\covfie\covfie\lib\core\covfie/core/algebra/vector.hpp(45,59): error C2059: syntax error: '...' [C:\Users\krasz\ATLAS\covfie\build\tests\core\test_core.vcxproj]
C:\Users\krasz\ATLAS\covfie\covfie\lib\core\covfie/core/algebra/vector.hpp(63,2): message : see reference to class template instantiation 'covfie::algebra::vector<N,T,I>' being compiled [C:\Users\krasz\ATLAS\covfie\build\tests\core\test_core.vcxproj]
C:\Users\krasz\ATLAS\covfie\covfie\lib\core\covfie/core/algebra/vector.hpp(48,19): error C2059: syntax error: '=' [C:\Users\krasz\ATLAS\covfie\build\tests\core\test_core.vcxproj]
C:\Users\krasz\ATLAS\covfie\covfie\lib\core\covfie/core/algebra/vector.hpp(50,9): error C2059: syntax error: ':' [C:\Users\krasz\ATLAS\covfie\build\tests\core\test_core.vcxproj]
C:\Users\krasz\ATLAS\covfie\covfie\lib\core\covfie/core/algebra/vector.hpp(50,1): error C2334: unexpected token(s) preceding ':'; skipping apparent function body [C:\Users\krasz\ATLAS\covfie\build\tests\core\test_core.vcxproj]
C:\Users\krasz\ATLAS\covfie\covfie\lib\core\covfie/core/algebra/vector.hpp(63): error C2143: syntax error: missing '>' before ';' [C:\Users\krasz\ATLAS\covfie\build\tests\core\test_core.vcxproj]
C:\Users\krasz\ATLAS\covfie\covfie\lib\core\covfie/core/algebra/vector.hpp(63): error C2059: syntax error: '>' [C:\Users\krasz\ATLAS\covfie\build\tests\core\test_core.vcxproj]
C:\Users\krasz\ATLAS\covfie\covfie\lib\core\covfie/core/algebra/vector.hpp(63): error C2238: unexpected token(s) preceding ';' [C:\Users\k rasz\ATLAS\covfie\build\tests\core\test_core.vcxproj]
C:\Users\krasz\ATLAS\covfie\covfie\lib\core\covfie/core/algebra/vector.hpp(63): fatal  error C1201: unable to continue after syntax error in class template definition [C:\Users\krasz\ATLAS\covfie\build\tests\core\test_core.vcxproj]
```

But in C\+\+20 mode it accepts the original code as well.

Since we were talking about this exact formalism in https://github.com/acts-project/algebra-plugins/pull/95 earlier yesterday, I tried that, and that worked with both C\+\+ standards...

Reply:
That's a shame, and quite odd since [fold expressions are a C++17 feature](https://en.cppreference.com/w/cpp/language/fold). Again, I don't mind _too_ much but it is a bit annoying that MSVC seems to refuse completely valid C++, and also I think the fold syntax is a lot clearer than the conjunction syntax... :frowning: